src/v0/destinations/iterable/deleteUsers.js

```diff
@@ -0,0 +1,78 @@
+const { httpDELETE } = require('../../../adapters/network');
+const { processAxiosResponse } = require('../../../adapters/utils/networkUtils');
+const { isHttpStatusSuccess } = require('../../util');
+const { getDynamicErrorType } = require('../../../adapters/utils/networkUtils');
+const { NetworkError, ConfigurationError } = require('../../util/errorTypes');
+const { executeCommonValidations } = require('../../util/regulation-api');
+const tags = require('../../util/tags');
+const { JSON_MIME_TYPE } = require('../../util/constant');
+
+// Ref-> https://developers.intercom.com/intercom-api-reference/v1.3/reference/permanently-delete-a-user
+const userDeletionHandler = async (userAttributes, config) => {
+  if (!config) {
+    throw new ConfigurationError('Config for deletion not present');
+  }
+  const { apiKey } = config;
+  if (!apiKey) {
+    throw new ConfigurationError('api key for deletion not present');
+  }
+  const validUserIds = [];
+  userAttributes.forEach((userAttribute) => {
+    // Dropping the user if userId is not present
+    if (userAttribute.userId) {
+      validUserIds.push(userAttribute.userId);
+    }
+  });
+  const failedUserDeletions = [];
+  await Promise.all(
+    validUserIds.map(async (uId) => {
+      const url = `https://api.iterable.com/api/users/byUserId/${uId}`;
+      const requestOptions = {
+        headers: {
+          'Content-Type': JSON_MIME_TYPE,
+          api_key: apiKey,
+        },
+      };
+      const resp = await httpDELETE(url, requestOptions, {
+        destType: 'iterable',
+        feature: 'deleteUsers',
+      });
+      const handledDelResponse = processAxiosResponse(resp);
+      if (!isHttpStatusSuccess(handledDelResponse.status) && handledDelResponse.status !== 404) {
+        if (handledDelResponse.status !== 400) {
+          throw new NetworkError(
```

Comment:
why do we need throw this?

Reply:
these are generic errors which will fail for all users, like invalid api key

Reply:
can you add comment for better clarity